components/script/dom/element.rs

```diff
@@ -1543,6 +1543,10 @@ impl ElementMethods for Element {
 
     // https://dom.spec.whatwg.org/#dom-element-setattributenode
     fn SetAttributeNode(&self, attr: &Attr) -> Fallible<Option<Root<Attr>>> {
+        // Workaround for https://github.com/servo/servo/issues/17366
+        // This ensures that if this is an "id" attr, its value is an Atom
```

Comment:
Well, to be honest is not really a workaround. It's the right fix, just not so efficient as it could be... Shrug.